src/Form/FooterLinkGeneralForm.php

```diff
@@ -53,28 +54,45 @@ public function form(array $form, FormStateInterface $form_state) {
 
   /**
    * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
-    $url = $form_state->getValue('url');
-    if (parse_url($url, PHP_URL_SCHEME) !== NULL) {
-      // We have nothing to validate on external URLs.
+    $uri = trim($form['url']['#value']);
+
+    if (parse_url($uri, PHP_URL_SCHEME) === NULL) {
+      if (strpos($uri, '<front>') === 0) {
+        $uri = '/' . substr($uri, strlen('<front>'));
+      }
+      $uri = 'internal:' . $uri;
+    }
+
+    // @see \Drupal\link\Plugin\Field\FieldWidget\LinkWidget::validateUriElement()
+    if (parse_url($uri, PHP_URL_SCHEME) === 'internal' &&
+      !in_array($form['url']['#value'][0], ['/', '?', '#'], TRUE) &&
+      substr($form['url']['#value'], 0, strlen('<front>')) !== '<front>') {
+      $form_state->setError($form['url'], t('The specified target is invalid. Manually entered paths should start with one of the following characters: / ? #'));
       return;
```

Comment:
It won't validate " /node/1" because of the starting space.
This is more simple and it ignores the space.
```

diff --git a/src/Form/FooterLinkGeneralForm.php b/src/Form/FooterLinkGeneralForm.php
index 9683bce..50a6b88 100644
--- a/src/Form/FooterLinkGeneralForm.php
+++ b/src/Form/FooterLinkGeneralForm.php
@@ -64,15 +64,14 @@ class FooterLinkGeneralForm extends EntityForm {
       if (strpos($uri, '<front>') === 0) {
         $uri = '/' . substr($uri, strlen('<front>'));
       }
-      $uri = 'internal:' . $uri;
-    }
 
-    // @see \Drupal\link\Plugin\Field\FieldWidget\LinkWidget::validateUriElement()
-    if (parse_url($uri, PHP_URL_SCHEME) === 'internal' &&
-      !in_array($form['url']['#value'][0], ['/', '?', '#'], TRUE) &&
-      substr($form['url']['#value'], 0, strlen('<front>')) !== '<front>') {
-      $form_state->setError($form['url'], t('The specified target is invalid. Manually entered paths should start with one of the following characters: / ? #'));
-      return;
+      // @see \Drupal\link\Plugin\Field\FieldWidget\LinkWidget::validateUriElement()
+      if (!in_array($uri[0], ['/', '?', '#'], TRUE)) {
+        $form_state->setError($form['url'], t('The specified target is invalid. Manually entered paths should start with one of the following characters: / ? #'));
+        return;
+      }
+
+      $uri = 'internal:' . $uri;
     }
 
     try {
```

Reply:
Problem with "<front>/node/1". It passes the validation and breaks the site.

Reply:
Fixed. The empty space is correctly not validating.